celery/contrib/conflator.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+"""celery.contrib.conflator
+======================
+
+.. versionadded :: 3.2.0
+
+A task which will check that it's enqueued time is after the last time
+a task with the same conflation ID was executed to clear backed up
+work that's already been done.
+
+This is useful for e.g. periodic tasks which occasionally take longer
+than their period to execute such as checking in with a remote system
+which may be down, experiencing performance issues or so on.
+
+It's important that things should work out satisfactorily at task
+execution time if all previously scheduled tasks execute or if they
+have all been conflated into 1.
+
+**Simple Example**
+
+.. code-block:: python
+
+    # sync from a remote system
+    @app.periodic_task(base=Conflator, run_every=timedelta(minutes=1))
+    def update_from_remote():
+        from remote_system import update
+        update()
+
+"""
+
+from datetime import datetime
+
+from celery import states, Task
+from celery.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+class Conflator(Task):
+
+    abstract = True
+
+    conflation_key = None
+
+    def apply_async(self, *args, **kwargs):
+        conflation_key = self.conflation_key
+        if not conflation_key:
```

Comment:
It is better to explicitly check `is None` when you want to check if it is None.